### PR TITLE
ensure < 6 levels of nesting scope #583

### PR DESCRIPTION
**Pull request by moldover**
Refactor the offending chunks of code into functions.

### CODE REVIEW

**Comment by moldover** [comment]
Btw, I could add the inspection file .idea/inspectionProfilesProject_Default.xml which does this but it looks like @HEdingfield git-ignored the inspections config file.  Was it causing problems with different versions or something?

**Review by HEdingfield — APPROVED** [review]
Hmm, I can't figure out which rule in there would exclude it. But, to answer your question, I think the idea was that our repo should be IDE-agnostic; we shouldn't assume that a developer is going to be using IntelliJ.

**Comment by tarheel** [comment]
I agree with @moldover -- if including this file is useful for anyone working in IntelliJ but it doesn't actually force a developer to use that IDE, then it's worth doing.

File: .gitignore [hunk 1]
```diff
@@ -26,15 +26,14 @@ tmp
...
-.idea/**/workspace.xml
-.idea/**/tasks.xml
-.idea/**/usage.statistics.xml
-.idea/**/dictionaries
-.idea/**/shelf
+#.idea/**/workspace.xml
+#.idea/**/tasks.xml
+#.idea/**/usage.statistics.xml
+#.idea/**/dictionaries
+#.idea/**/shelf
```
[the remark this answers is not in the document]
Comment (moldover) [reply .gitignore]: Yeah agreed.  I put em back.

**Pull request merged**